src/core/libmaven/PeakDetector.cpp

```diff
@@ -316,41 +461,47 @@ void PeakDetector::processSlices(vector<mzSlice*> &slices, string setName)
                             slice,
                             mp,
                             PeakGroup::IntegrationType::Automated);
-        GroupFiltering groupFiltering(mavenParameters, slice);
-        groupFiltering.filter(peakgroups);
-
-        // sort groups according to their rank
-        std::sort(peakgroups.begin(), peakgroups.end(),
-                  PeakGroup::compRank);
 
-        for (unsigned int j = 0; j < peakgroups.size(); j++) {
-            // check for duplicates	and append group
-            if (j >= mavenParameters->eicMaxGroups)
-                break;
-            mavenParameters->allgroups.push_back(peakgroups[j]);
+        // we do not filter non-parent adducts or non-parent isotopologues
+        bool isParentGroup = slice->adduct == nullptr
```

Comment:
To self:
What is the `slice->adduct` value when it's an isotope slice?

Reply:
What is this "to self" thing?

All slices that have a target compound have a non-null adduct. Isotope slices have a target compound, so their adduct is by default [M+H] or [M-H] depending on global polarity.

Reply:
I was writing a note for myself to go back and check. Forgot I wasn't in review mode so it got published.